topdown/http.go

```diff
@@ -152,6 +188,35 @@ func executeHTTPRequest(bctx BuiltinContext, obj ast.Object) (ast.Value, error)
 		}
 	}
 
+	if tlsClientCertFile != "" && tlsClientKeyFile != "" {
+		clientCertFromFile, err := tls.LoadX509KeyPair(tlsClientCertFile, tlsClientKeyFile)
+		if err != nil {
+			return nil, err
+		}
+		clientCerts = append(clientCerts, clientCertFromFile)
+	}
+
+	if len(tlsClientCertEnvVar) > 0 && len(tlsClientKeyEnvVar) > 0 {
+		clientCertFromEnv, err := tls.X509KeyPair(tlsClientCertEnvVar, tlsClientKeyEnvVar)
+		if err != nil {
+			return nil, err
+		}
+		clientCerts = append(clientCerts, clientCertFromEnv)
+	}
+
+	if len(clientCerts) > 0 {
+		// this is a TLS connection
+		connRootCAs, err := createRootCAs(tlsCaCertFile, tlsCaCertEnvVar, tlsUseSystemCerts)
+		if err != nil {
+			return nil, err
+		}
+		tlsConfig.Certificates = append(tlsConfig.Certificates, clientCerts...)
+		tlsConfig.RootCAs = connRootCAs
+		client.Transport = &http.Transport{
```

Comment:
⚠️ Let's be careful with this. The zero values for `http.Transport` can cause a connection leak if used with many different target hosts. (See https://github.com/open-policy-agent/opa/commit/c77f9ed228a224a847c07eb1d2da90a176f5c666.)

But, I think this might not matter since we've got a fresh `http.Transport` for each request anyways... I'm not sure... just raising awareness. It's _probably fine_.